From 83b49400c1dfb72e3019267fd03f165d0bffec76 Mon Sep 17 00:00:00 2001
From: Fiona Klute <fiona.klute@gmx.de>
Date: Mon, 6 Oct 2025 15:20:26 +0200
Subject: [PATCH] build: Allow libdisplay-info 0.3.0

This is a backport of the change in upstream commit
683eabb61ec43a270f2e3ed67c63bbe3754b1214 "build: Allow libdisplay-info 0.3.0"
[1], applying the loosened version limit to the DRM backend only. The dependency
definition was moved to the top level meson.build in [2], and the lower bound
increased in [3].

[1] https://gitlab.freedesktop.org/wayland/weston/-/commit/683eabb61ec43a270f2e3ed67c63bbe3754b1214
[2] https://gitlab.freedesktop.org/wayland/weston/-/commit/76e659f45edd23a8b1b253a93316ec73c709bc71
[3] https://gitlab.freedesktop.org/wayland/weston/-/commit/82c5b386d860a479ed5a157f4dde58ffe202b948

Upstream: https://gitlab.freedesktop.org/wayland/weston/-/commit/683eabb61ec43a270f2e3ed67c63bbe3754b1214 (backported)
Signed-off-by: Fiona Klute (WIWA) <fiona.klute@gmx.de>
---
 libweston/backend-drm/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/backend-drm/meson.build b/libweston/backend-drm/meson.build
index 002669e6..a8f2ea59 100644
--- a/libweston/backend-drm/meson.build
+++ b/libweston/backend-drm/meson.build
@@ -4,7 +4,7 @@ endif
 
 dep_libdisplay_info = dependency(
 	'libdisplay-info',
-	version: ['>= 0.1.1', '< 0.3.0'],
+	version: ['>= 0.1.1', '< 0.4.0'],
 	fallback: ['display-info', 'di_dep'],
 	default_options: [
 		'werror=false',
-- 
2.51.0

